### PR TITLE
Port kernelMachine to openmp

### PR DESCRIPTION
**Pull request by MikeLing**

### CODE REVIEW

File: src/shogun/machine/KernelMachine.cpp [hunk 1]
```diff
@@ -553,72 +503,57 @@ SGVector<float64_t> CKernelMachine::apply_locked_get_output(
...
+	#pragma omp parallel shared(num_threads, step)
 	{
-		pthread_t* threads = SG_MALLOC(pthread_t, num_threads-1);
-		S_THREAD_PARAM_KERNEL_MACHINE* params=SG_MALLOC(S_THREAD_PARAM_KERNEL_MACHINE, num_threads);
-		int32_t step= num_inds/num_threads;
-
-		int32_t t;
-		for (t=0; t<num_threads-1; t++)
+#ifdef HAVE_OPENMP
+		#pragma opm single
```
Comment (lisitsyn) [inline src/shogun/machine/KernelMachine.cpp]: Does that compile? Typo here

Comment (MikeLing) [inline src/shogun/machine/KernelMachine.cpp]: mmm, I think so. I compile it under the environment with openmp. I don't know why there's no error comes out. I will fix it right now :)

**Comment by MikeLing** [comment]
Ask review for this pr

File: src/shogun/machine/KernelMachine.cpp [hunk 2]
```diff
@@ -333,64 +337,62 @@ SGVector<float64_t> CKernelMachine::apply_get_outputs(CFeatures* data)
...
-			{
-				pthread_t* threads = SG_MALLOC(pthread_t, num_threads-1);
-				S_THREAD_PARAM_KERNEL_MACHINE* params = SG_MALLOC(S_THREAD_PARAM_KERNEL_MACHINE, num_threads);
-				int32_t step= num_vectors/num_threads;
-
-				int32_t t;
 
-				for (t=0; t<num_threads-1; t++)
+#ifdef HAVE_OPENMP
+#pragma opm single
```
Comment (vigsterkr) [inline src/shogun/machine/KernelMachine.cpp]: typo.... due to this as it was mentioned by @karlnapf in https://github.com/shogun-toolbox/shogun/pull/3738#issuecomment-289268657 that be great that our unit test somehow tests both multi and single threaded scenario...

File: src/shogun/machine/KernelMachine.cpp [hunk 3]
```diff
@@ -553,72 +507,60 @@ SGVector<float64_t> CKernelMachine::apply_locked_get_output(
...
-	else
-	{
-		pthread_t* threads = SG_MALLOC(pthread_t, num_threads-1);
-		S_THREAD_PARAM_KERNEL_MACHINE* params=SG_MALLOC(S_THREAD_PARAM_KERNEL_MACHINE, num_threads);
-		int32_t step= num_inds/num_threads;
-
-		int32_t t;
-		for (t=0; t<num_threads-1; t++)
+#ifdef HAVE_OPENMP
+#pragma opm single
```
Comment (vigsterkr) [inline src/shogun/machine/KernelMachine.cpp]: same typo here...

File: src/shogun/machine/KernelMachine.cpp (hunk 2 above)
Comment (MikeLing) [inline src/shogun/machine/KernelMachine.cpp]: I found the comment in https://github.com/shogun-toolbox/shogun/pull/3738#issuecomment-289267398 doesn't really helps to test multi threaded situation. `omp_get_num_threads()` always return 1 even I export OMP_NUM_THREADS as 4. How could I validate the openmp really works?

Comment (MikeLing) [inline src/shogun/machine/KernelMachine.cpp]: I just use `parallel->get_num_threads()` to get the number of threads, but the result stay the same (single thread) :/

**Pull request merged**